roles/hadoop-install/tasks/main.yml

```diff
@@ -1,5 +1,24 @@
 ---
 
+  - name: hadoop | service | hdfs | logic flags | compute
+    set_fact:
+        remove_data_root: "{{ state == 'absent' }}"
+        remove_install_root: "{{ state == 'absent' }}"
+        stop_hdfs_services: >
+            {{ (state == 'absent' or (format|bool)) and
+               inventory_hostname in
+                   groups.get(hdfs_namenode_ansible_group, []) }}
+        do_install: >
+            {{state == "present" or state == "stopped" or
+              state == "started" or state == "restarted" or
+              state == "reloaded"}}
```

Comment:
@cpatrick If you think it's just a boolean value that controls whether the tasks that install hadoop should be run, then you are correct.  :)

The `|bool` part is just an ansible filter that turns string values "True", "true", "yes", and possibly others to the boolean value `True` (and their negative counterparts to `False`).  Since there's no way to return a non-string from an expression like the one here, we have to use the `bool` filter at evaluation time.